packages/block-library/src/table/edit.js

```diff
@@ -0,0 +1,295 @@
+			this.setState( { selectedCell: null } );
+		};
+	}
+
+	createOnDeleteRow( { section, rowIndex } ) {
+		const { attributes, setAttributes } = this.props;
+
+		return () => {
+			setAttributes( deleteRow( attributes, { section, rowIndex } ) );
+			this.setState( { selectedCell: null } );
+		};
+	}
+
+	createOnInsertColumn( { section, columnIndex }, delta ) {
+		const { attributes, setAttributes } = this.props;
+
+		columnIndex += delta;
+
+		return () => {
+			setAttributes( insertColumn( attributes, { section, columnIndex } ) );
+			this.setState( { selectedCell: null } );
+		};
+	}
+
+	createOnDeleteColumn( { section, columnIndex } ) {
+		const { attributes, setAttributes } = this.props;
+
+		return () => {
+			setAttributes( deleteColumn( attributes, { section, columnIndex } ) );
+			this.setState( { selectedCell: null } );
+		};
+	}
+
+	createOnFocus( selectedCell ) {
+		return () => {
+			this.setState( { selectedCell } );
+		};
+	}
+
+	getTableControls() {
+		const { selectedCell } = this.state;
+
+		return [
+			{
+				icon: 'table-row-before',
+				title: __( 'Add Row Before' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertRow( selectedCell ),
+			},
+			{
+				icon: 'table-row-after',
+				title: __( 'Add Row After' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertRow( selectedCell, 1 ),
+			},
+			{
+				icon: 'table-row-delete',
+				title: __( 'Delete Row' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnDeleteRow( selectedCell ),
+			},
+			{
+				icon: 'table-col-before',
+				title: __( 'Add Column Before' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertColumn( selectedCell ),
+			},
+			{
+				icon: 'table-col-after',
+				title: __( 'Add Column After' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertColumn( selectedCell, 1 ),
+			},
+			{
+				icon: 'table-col-delete',
+				title: __( 'Delete Column' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnDeleteColumn( selectedCell ),
+			},
+		];
+	}
+
+	renderSection( { type, rows } ) {
+		if ( ! rows.length ) {
+			return null;
+		}
+
+		const Tag = `t${ type }`;
+		const { selectedCell } = this.state;
+
+		return (
+			<Tag>
+				{ rows.map( ( { cells }, rowIndex ) =>
+					<tr key={ rowIndex }>
+						{ cells.map( ( { content, tag: CellTag }, columnIndex ) => {
+							const isSelected = selectedCell && (
+								type === selectedCell.section &&
+								rowIndex === selectedCell.rowIndex &&
+								columnIndex === selectedCell.columnIndex
+							);
+
+							const id = {
+								section: type,
+								rowIndex,
+								columnIndex,
+							};
+
+							const classes = classnames( {
+								'is-selected': isSelected,
+							} );
+
+							return (
+								<CellTag key={ columnIndex } className={ classes }>
+									<RichText
```

Comment:
A 10x10 table means we will have 100 instances of TinyMCE loaded on the page. Will these sorts of uses-cases use too much memory?

Reply:
Yes, sure, we should also be looking into the general performance of this. A table with 100 cells is the same as 100 paragraphs.